edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Service/DependencyManager.cs

```diff
@@ -162,6 +162,7 @@ void RegisterMqttModule(ContainerBuilder builder, (bool isEnabled, bool usePersi
             TimeSpan configUpdateFrequency = TimeSpan.FromSeconds(configUpdateFrequencySecs);
             bool checkEntireQueueOnCleanup = this.configuration.GetValue("CheckEntireQueueOnCleanup", false);
             bool closeCloudConnectionOnDeviceDisconnect = this.configuration.GetValue("CloseCloudConnectionOnDeviceDisconnect", true);
+            bool isLegacyUpstream = this.configuration.GetValue("mqttBrokerSettings:legacyUpstream", true);
```

Comment:
>bool isLegacyUpstream = this.configuration.GetValue("mqttBrokerSettings:legacyUpstream", true); [](start = 12, length = 95)

this should be derived from a combination of (mqttBrokerEnabled && nestedEdgeEnabled && IOTEDGE_GATEWAYHOSTNAME != null)